### ISSUE TITLE
Crash when trying to share IPP feedback

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#8791]
The app crashes (in TestFlight beta v. 12.1) when I tap the "Share Feedback" button on the prompt to share IPP feedback on the Orders tab. Here's the relevant part of the stacktrace:

```
0   libswiftCore.dylib            	0x000000018c6d0d7c _assertionFailure(_:_:file:line:flags:) + 312 (AssertCommon.swift:143)
1   WooCommerce                   	0x00000001030832d8 static WooConstants.URLs.trustedURL(_:) + 372 (WooConstants.swift:276)
2   WooCommerce                   	0x0000000102ca6f94 WooConstants.URLs.asURL() + 28 (WooConstants.swift:262)
3   WooCommerce                   	0x0000000102ca6f94 SurveyViewController.Source.url.getter + 2760 (SurveyViewController.swift:115)
4   WooCommerce                   	0x0000000102ca634c SurveyViewController.configureAndLoadSurvey() + 324 (SurveyViewController.swift:54)
5   WooCommerce                   	0x0000000102ca61f0 0x10295c000 + 3449328 (SurveyViewController.swift:48)
6   WooCommerce                   	0x0000000102ca64bc @objc SurveyViewController.viewDidLoad() + 28 (<compiler-generated>:0)
```

As recorded in Sentry:

> WooCommerce/WooConstants.swift:276: Fatal error: Expected URL https://automattic.survey.fm/woo-app-–-ipp-first-transaction-survey to be a well-formed URL.

cc @iamgabrielma

**Comment by iamgabrielma** [comment]
Thanks for the ping! By any chance, do you recall if the banner had any "message", or just the title and the feedback button? 
Screenshots below

| Banner with info message | Banner without info message |
|--|--|
| <img width="415" alt="Screenshot 2023-02-01 at 21 44 47" src="https://user-images.githubusercontent.com/3812076/216059917-8bc664c8-2c18-41bb-b610-d915675f9eea.png"> | <img width="400" alt="Screenshot 2023-02-01 at 21 43 44" src="https://user-images.githubusercontent.com/3812076/216059938-db2729cb-9875-402c-b6a1-60ba319c05fb.png"> |

I'm thinking it could be related to [this edge case](https://github.com/woocommerce/woocommerce-ios/pull/8748) when sometimes the message doesn't appear, if we don't have time to fetch the banner message, it could be possible that we couldn't fetch the link either. 

I'll test this further and either provide a fallback link, or hide it entirely if the message doesn't load on time.

**Comment by rachelmcr** [comment]
I'm not sure which version I saw. 🤔  But the correct survey URL was included in the fatal error message logged in Sentry, so I don't think a missing link is the problem. (`Fatal error: Expected URL [https://automattic.survey.fm/woo-app-–-ipp-first-transaction-survey] to be a well-formed URL.`)

It looks like the problem is with one of the characters in the URL string not being escaped. When it isn't escaped, it can't be turned into a URL, whereas when it is escaped it works fine:

```swift
// This is `nil`
let brokenUrl = URL(string: "https://automattic.survey.fm/woo-app-–-ipp-first-transaction-survey")

// This is the expected optional URL
let workingUrl = URL(string: "https://automattic.survey.fm/woo-app-%E2%80%93-ipp-first-transaction-survey")
```

**Comment by iamgabrielma** [comment]
Jinx! I just dropped a message on p1675262943438489-C025A8VV728 with what seems to be the issue. One of the `-` in the URL uses a different character indeed.

**Comment by iamgabrielma** [comment]
Resolved with #8805. Closing!
